File: core/wifi_manager.py
import os
import clr
import sys
from typing import List, Optional

# Подключаем твою скомпилированную DLL
dll_path = os.path.abspath("WifiCore.dll")
if os.path.exists(dll_path):
    try:
        clr.AddReference(dll_path)
        from WifiCore import WifiManager as InternalWifiManager
    except Exception as e:
        print(f"Ошибка загрузки WifiCore.dll: {e}")
        InternalWifiManager = None
else:
    InternalWifiManager = None

class WindowsWifiManager:

    def scan_networks(self, event=None) -> List[str]:
        """Псевдоним для совместимости с GUI и обработки события нажатия кнопки."""
        return self.scan_results()

    def __init__(self):
        self._manager = InternalWifiManager() if InternalWifiManager else None
        # GUI ожидает список имен при инициализации
        self.interfaces = self.get_interface_names()

    def get_interface_names(self) -> List[str]:
        if not self._manager:
            return ["Ядро не загружено"]
        try:
            # Вызываем метод из твоего C# кода
            return list(self._manager.GetInterfaceNames())
        except Exception as e:
            return [f"Ошибка API: {e}"]

    # Для совместимости, если где-то вызывается старое имя
    def get_interfaces(self) -> List[str]:
        return self.get_interface_names()

    def scan_results(self) -> List[str]:
        if not self._manager:
            return ["Ошибка: DLL не найдена"]
        return list(self._manager.ScanNetworks())

    def get_profiles(self) -> List[str]:
        if not self._manager:
            return []
        return list(self._manager.GetProfiles())

    def get_password(self, profile_name: str) -> str:
        if not self._manager:
            return "Ошибка"
        return self._manager.GetProfilePassword(profile_name)

    def get_interface_by_index(self, index: int) -> Optional[str]:
        if 0 <= index < len(self.interfaces):
            return self.interfaces[index]
        return None

File: WifiCore_CS/WifiManager.cs
# core/wifi_manager.py
import clr
import sys
import os
from typing import List, Optional

# Добавляем путь к нашей DLL
dll_path = os.path.join(os.path.dirname(__file__), '..', 'WifiCore.dll')
clr.AddReference(dll_path)

# Импортируем наш C# класс
from WifiCore import WifiManager

class WindowsWifiManager:
    """Обёртка для C# класса WifiManager."""
    
    def __init__(self):
        try:
            self.wifi_core = WifiManager()
            print("[+] C# ядро WifiManager успешно загружено")
        except Exception as e:
            print(f"[-] Ошибка загрузки C# ядра: {e}")
            self.wifi_core = None
    
    def get_interface_names(self) -> List[str]:
        """Возвращает список описаний WiFi интерфейсов (заменяет старый метод)."""
        if not self.wifi_core:
            return []
        return list(self.wifi_core.GetInterfaceNames())
    
    def scan_networks(self, interface_index: int = 0) -> List[str]:
        """Сканирует сети на выбранном интерфейсе (interface_index пока игнорируем)."""
        if not self.wifi_core:
            return []
        return list(self.wifi_core.ScanNetworks())
    
    # ИСПРАВЛЕНИЕ: Переименовываем метод для соответствия GUI
    def get_saved_profiles(self, interface_index: int = 0) -> List[str]:
        """Возвращает список сохранённых WiFi профилей (interface_index игнорируем)."""
        if not self.wifi_core:
            return []
        return list(self.wifi_core.GetProfiles())
    
    # Новый метод для получения пароля (для будущего использования)
    def get_profile_password(self, profile_name: str) -> str:
        """Возвращает пароль для указанного профиля."""
        if not self.wifi_core:
            return "Ядро не загружено"
        return self.wifi_core.GetProfilePassword(profile_name)

# Быстрый тест
if __name__ == "__main__":
    manager = WindowsWifiManager()
    print("Интерфейсы:", manager.get_interface_names())
    print("\nСканирование сетей...")
    networks = manager.scan_networks()
    for net in networks[:3]:
        print(" ", net)
    print("\nПрофили:")
    profiles = manager.get_saved_profiles()
    for profile in profiles[:5]:
        print(" ", profile)